mucommander-core/src/main/java/com/mucommander/ui/action/impl/ShowTerminalAction.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.ui.action.impl;
+
+import com.jediterm.terminal.ui.JediTermWidget;
+import com.mucommander.commons.file.AbstractFile;
+import com.mucommander.commons.file.protocol.local.LocalFile;
+import com.mucommander.core.desktop.DesktopManager;
+import com.mucommander.text.Translator;
+import com.mucommander.ui.action.AbstractActionDescriptor;
+import com.mucommander.ui.action.ActionCategory;
+import com.mucommander.ui.action.ActionDescriptor;
+import com.mucommander.ui.action.ActionProperties;
+import com.mucommander.ui.main.MainFrame;
+
+import java.awt.BorderLayout;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.util.Map;
+
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+
+/**
+ * This action shows built-in terminal (it mimics the behavior of Midnight Commander Ctrl-O command
+ * that originates back from Norton Commander).
+ */
+public class ShowTerminalAction extends ActiveTabAction {
```

Comment:
maybe it's time to add an osgi extension point for actions? that's something I thought about long time ago but didn't file a good justification but here I see that the functionality can be added as a module that register an action, what do you think?

Reply:
Sounds good, would it mean that terminal and those import could be in a separate subsystem/module?

Reply:
yep, separate osgi bundle :)